application/libraries/PluginManager/LimesurveyApi.php

```diff
@@ -473,4 +473,90 @@ public function renderTwig($file, array $data)
             false
         );
     }
+
+    /**
+     * Returns an array of all user groups
+     *
+     * @return array
+     */
+    public function getUserGroups()
+    {   
+        return \UserGroup::model()->findAllAsArray();
+    }
+
+    /**
+     * Adds a new user group
+     *
+     * @param string $groupName Name of user group to be created
+     * @param string $groupDescription Description of user group to be created
+     * @return boolean True or false if user group was added or not
+     */
+    public function addUserGroup($groupName, $groupDescription)
+    {
+        $db_group_name = flattenText($groupName, false, true, 'UTF-8', true);
+        $db_group_description = $groupDescription;
+
+        if (isset($db_group_name) && strlen($db_group_name) > 0) {
+            if (strlen($db_group_name) > 21) {
+                return false;
+            } elseif (\UserGroup::model()->find("name=:groupName", array(':groupName'=>$db_group_name))) {
+                return false;
+            } else {
+                $newUserGroup = new \UserGroup();
+                $newUserGroup->owner_id = 1;
+                $newUserGroup->name = $groupName;
+                $newUserGroup->description = $groupDescription;
+                $result = $newUserGroup->save();
+                if ($result) {
+                    \UserInGroup::model()->insertRecords(array('ugid' => $newUserGroup->getPrimaryKey(), 'uid' => 1));
+                }
+                return $result;
+            }
+        } else {
+            return false;
+        }
+    }
+
+    /**
+     * Adds or removes a user from a user group
+     *
+     * @param integer $ugid The user group ID
+     * @param integer $uid The user ID
+     * @param string $action The desired action 'add' or 'remove'
+     * @return boolean True or false if user was added/removed to/from group or not
+     */
+    public function userInGroupAction($ugid, $uid, $action)
```

Comment:
Please remove `$action` and split the function in two parts, one for remove, one for add. If the functions have code in common, split it into a third private function.